ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -695,25 +707,73 @@ private Dataset executeGetDatasetStatement(PreparedStatement statement, int acco
         retentionCount, userTags);
   }
 
+  /**
+   * Get version schema corresponding to the dataset.
+   * @param statement the mysql statement to get the version schema of the dataset.
+   * @param accountId the id of the parent account.
+   * @param containerId the if of the container.
+   * @param datasetName the name of the dataset.
+   * @return the name of the dataset.
+   * @throws SQLException
+   */
+  private Dataset.VersionSchema executeGetVersionSchema(PreparedStatement statement, int accountId, int containerId,
+      String datasetName) throws SQLException {
+    ResultSet resultSet = null;
+    Dataset.VersionSchema versionSchema;
+    try {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, datasetName);
+      resultSet = statement.executeQuery();
+      resultSet.next();
+      versionSchema = Dataset.VersionSchema.values()[resultSet.getInt(VERSION_SCHEMA)];
```

Comment:
is it possible that dataset doesn't exist , so the resultSet doesn't exist.

Reply:
Yes, I updated the one related to schema version query, I have another pr which fixed the rest of get queries.